systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java

```diff
@@ -1069,6 +1075,333 @@ void testRestoreOffsetsInConsumerGroup(ExtensionContext extensionContext) {
+        mmSnapshot = DeploymentUtils.waitTillDepHasRolled(namespaceName, KafkaMirrorMaker2Resources.deploymentName(clusterName), 1, mmSnapshot);
+        Map<String, String> kafkaSourcePods = StatefulSetUtils.ssSnapshot(namespaceName, KafkaResources.kafkaStatefulSetName(kafkaClusterSourceName));
+        Map<String, String> kafkaTargetPods = StatefulSetUtils.ssSnapshot(namespaceName, KafkaResources.kafkaStatefulSetName(kafkaClusterSourceName));
+
+        LOGGER.info("[SCRAM-SHA] Restart kafkaClients pod with new passwords.");
+        resourceManager.deleteResource(kubeClient().namespace(namespaceName).getDeployment(kafkaClientsName));
+        resourceManager.createResource(extensionContext, KafkaClientsTemplates.kafkaClients(namespaceName, true, kafkaClientsName, userSource, userTarget).build());
+        kafkaClientsPodName = kubeClient().listPodsByPrefixInName(kafkaClientsName).get(0).getMetadata().getName();
+
+        internalKafkaClient = internalKafkaClient.toBuilder()
+                .withUsingPodName(kafkaClientsPodName)
+                .withTopicName(topicSourceName)
+                .withClusterName(kafkaClusterSourceName)
+                .withKafkaUsername(userSource.getMetadata().getName())
+                .withListenerName(kafkaTlsScramListener)
+                .build();
+
+        LOGGER.info("[SCRAM-SHA] Sending messages to - topic {}, cluster {} and message count of {}",
+                topicSourceName, kafkaClusterSourceName, messagesCount);
+        sent = internalKafkaClient.sendMessagesTls();
+
+        LOGGER.info("[SCRAM-SHA] Changing to target - topic {}, cluster {}, user {}", topicTargetName, kafkaClusterTargetName, userTarget.getMetadata().getName());
+        internalKafkaClient = internalKafkaClient.toBuilder()
+                .withTopicName(topicTargetName)
+                .withClusterName(kafkaClusterTargetName)
+                .withKafkaUsername(userTarget.getMetadata().getName())
+                .build();
+
+        LOGGER.info("[SCRAM-SHA] Now messages should be mirrored to target topic and cluster");
+        internalKafkaClient.checkProducedAndConsumedMessages(sent, internalKafkaClient.receiveMessagesTls());
+        LOGGER.info("SCRAM-SHA] Messages successfully mirrored");
+
+        LOGGER.info("[TLS] Change CA cert for Source clusters by forcing annotation");
+        String clusterCaSecretName = KafkaResources.clusterCaCertificateSecretName(kafkaClusterSourceName);
+        Secret secret = new SecretBuilder(kubeClient(namespaceName).getSecret(namespaceName, clusterCaSecretName))
+            .editMetadata()
+                .addToAnnotations(Ca.ANNO_STRIMZI_IO_FORCE_RENEW, "true")
+            .endMetadata().build();
+        kubeClient(namespaceName).patchSecret(namespaceName, clusterCaSecretName, secret);
+        kafkaSourcePods = StatefulSetUtils.waitTillSsHasRolled(KafkaResources.kafkaStatefulSetName(kafkaClusterTargetName), 3, kafkaSourcePods);
```

Comment:
You are waiting for rolling update of `kafkaClusterTargetName`, then the snapshot you are loading to `kafkaSourcePods` - this is wrong.

Reply:
Yeah, these are copy/paste errors. Sadly test is passing even with them :-(

Reply:
It can locally pass in case of some race condition, or if the second cluster should roll also, it will wait there.